lib/viewcomponents/vertex.jsx

```diff
@@ -155,8 +155,10 @@ var Vertex = React.createClass({
     var vPct = pos.y/styles.get('height');
 
     var cardinalPosition = PositionUtils.Conversion.cartesianToCardinal({x: hPct, y: vPct});
-
-    portModel.set('position', cardinalPosition);
+    portModel.setAttrs({
+        isPinned: true,
+		position: cardinalPosition
+      });
```

Comment:
That looks super janky and not at all how it looks on my machine lol but I'll take a look maybe its the editor